lldb/tools/lldb-dap/Handler/ConfigurationDoneRequestHandler.cpp

```diff
@@ -52,8 +64,14 @@ void ConfigurationDoneRequestHandler::operator()(
   dap.configuration_done_sent = true;
   if (dap.stop_at_entry)
     SendThreadStoppedEvent(dap);
-  else
+  else {
+    // Client requests the baseline of currently existing threads after
+    // a successful launch or attach.
+    // A 'threads' request will be sent after configurationDone response
+    // Obtain the list of threads before we resume the process
```

Comment:
Maybe expand on this a bit saying something like:
```
// If a 'threads' request returns no threads, then the interrupt 
// button won't be available in the GUI debugger controls, so
// we need to report valid threads back to the IDE. We also need
// a valid thread for the "continued" packet to be sent.
```

Reply:
The pause button relies on having some threads informations and it always send the `threadID` in the pause request. I have also checked the [VSCode source code](https://github.com/microsoft/vscode/blob/0b0be005fb90314a4659b7ca2a9c5d5dd2dd4378/src/vs/workbench/contrib/debug/browser/debugSession.ts#L1041) about fetching threads. If no threads are returned in the threads response then VSCode won't update the know thread list.

Which means, we really just need to return the baseline of threads for the debug session to carry on.

Reply:
As for the `continued` event, VSCode will ignore the threadID if `allThreadsContinued: true`. Looks like can send an invalid threadID if resuming all threads

Reply:
yeah, we just need to change our code to always send it even if the `dap.focus_tid` is not valid.